src/sentry/eventstream/kafka/protocol.py

```diff
@@ -104,59 +104,65 @@ def get_task_kwargs_for_message_from_headers(headers: Sequence[Tuple[str, Option
     Same as get_task_kwargs_for_message but gets the required information from
     the kafka message headers.
     """
-    try:
-        header_data = {k: v for k, v in headers}
-        if "group_id" not in header_data:
-            header_data["group_id"] = None
-        if "primary_hash" not in header_data:
-            header_data["primary_hash"] = None
 
-        def decode_str(value: Optional[bytes]) -> str:
-            assert isinstance(value, bytes)
-            return value.decode("utf-8")
+    def decode_str(value: Optional[bytes]) -> str:
+        assert isinstance(value, bytes)
+        return value.decode("utf-8")
 
-        def decode_optional_str(value: Optional[bytes]) -> Optional[str]:
-            if value is None:
-                return None
-            return decode_str(value)
+    def decode_optional_str(value: Optional[bytes]) -> Optional[str]:
+        if value is None:
+            return None
+        return decode_str(value)
 
-        def decode_int(value: Optional[bytes]) -> int:
-            assert isinstance(value, bytes)
-            return int(value)
+    def decode_int(value: Optional[bytes]) -> int:
+        assert isinstance(value, bytes)
+        return int(value)
 
-        def decode_optional_int(value: Optional[bytes]) -> Optional[int]:
-            if value is None:
-                return None
-            return decode_int(value)
+    def decode_optional_int(value: Optional[bytes]) -> Optional[int]:
+        if value is None:
+            return None
+        return decode_int(value)
 
-        def decode_bool(value: bytes) -> bool:
-            return bool(int(decode_str(value)))
+    def decode_bool(value: bytes) -> bool:
+        return bool(int(decode_str(value)))
```

Comment:
These are nitpicks/micro optimizations, but I wonder if it would be faster to do `return value == b"1"` instead of doing multiple casts.

Reply:
There are certainly more efficient ways to do things but I think I'm going to leave this for my own sanity.

Reply:
I hope we can find bigger optimisations first before resorting to these.